src/effects/builtin/tonegen/toneeffect.cpp

```diff
@@ -12,6 +12,10 @@ ToneEffect::ToneEffect()
     : ToneGenBase(false), GeneratorEffect(mT0, mT1)
 {}
 
+ToneEffect::ToneEffect(bool isChirp)
```

Comment:
Why not simplify using just one ctor with bool (or enum) ?

Reply:
It is instantiated by `BuiltinEffectsModule::Registration` which does not allow us to pass arguments to ctor.

Reply:
We could use a templated class instead, though not sure if it make the code prettier.